docs/csharp/versioning.md

```diff
@@ -0,0 +1,132 @@
+---
+title: C# Versioning | C# Guide
+description: Understand how versioning works in C# and .NET
+keywords: .NET, .NET Core, C#
+author: tsolarin
+manager: wpickett
+ms.date: 2016/12/03
+ms.topic: article
+ms.prod: visual-studio-dev-14
+ms.technology: devlang-csharp
+ms.devlang: csharp
+ms.assetid: aa8732d7-5cd0-46e1-994a-78017f20d861
+---
+
+# Versioning in C# #
+
+In this tutorial you'll learn what versioning means in .NET. You'll also learn the factors to consider when versioning your library as well as upgrading
+to a new version of the a library.
+
+## Authoring Libraries
+
+As a developer who has created .NET libraries for public use, you've most likely been in situations where you have
+to roll out new updates. How you go about this process matters a lot as you need to ensure that there's a seamless transition
+of existing code to the new version of your library. Here are several things to consider when creating a new release:
+
+### Semantic Versioning
+
+[Semantic versioning](http://semver.org/) (SemVer for short) is a naming convention applied to versions of your library to signify specific milestone events.
+Ideally, the version information you give your library should help developers determine the compatibility
+with their projects that make use of older versions of that same library.
+
+The most basic approach to SemVer is the 3 component format `MAJOR.MINOR.PATCH`, where:
+ 
+* `MAJOR` is incremented when you make incompatible API changes
+* `MINOR` is incremented when you add functionality in a backwards-compatible manner
+* `PATCH` is incremented when you make backwards-compatible bug fixes
+
+There are also ways to specify other scenarios like pre-release versions etc. when applying version information to your .NET library.
+
+### Backwards Compatibility
+
+As you release new versions of your library, backwards compatibility with previous versions will most likely be one of your major concerns.
+A new version of your library is source compatible with a previous version if code that depends on the previous version can, when recompiled, work with the new version. 
+A new version of your library is binary compatible if an application that depended on the old version can, without recompilation, work with the new version.
+
+Here are some things to consider when trying to maintain backwards compatibility with older versions of your library:
+
+* Virtual methods: When you make a virtual method non-virtual in your new version it means that projects that override that method
```

Comment:
I'd be more prescriptive here. I think it is safe to say that is such a breaking change that this should be strongly discouraged.